lib/fog/aws/storage.rb

```diff
@@ -256,21 +256,25 @@ def request_params(params)
           if params[:bucket_name]
             bucket_name = params[:bucket_name]
 
-            path_style = params.fetch(:path_style, @path_style)
-            if !path_style
-              if COMPLIANT_BUCKET_NAMES !~ bucket_name
-                Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) is not a valid dns name, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                path_style = true
-              elsif scheme == 'https' && !path_style && bucket_name =~ /\./
-                Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) contains a '.' so is not accessible over https as a virtual hosted bucket, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                path_style = true
+            if params[:bucket_cname]
+              host = bucket_name
```

Comment:
@geemus I'm using a bucket_name such as: `assets.example.com`. As per this [S3 documentation page](http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html#VirtualHostingCustomURLs), you can then have functioning urls in the form of: `https://assets.example.com/the/path/to.png`

I have only been able to get an url in that form by using this patch and passing in this option, but glad to be shown other ways of achieving this.